### PR TITLE
Enforce naming pattern for version catalogs

### PR DESCRIPTION
**Pull request by ljacomet**
In order to reduce extension collisions, enforce that a catalog nmae
must end with 'Libs'

### CODE REVIEW

**Comment by ljacomet** [comment]
@bot-gradle test this

File: subprojects/dependency-management/src/main/java/org/gradle/internal/management/DefaultVersionCatalogBuilderContainer.java [hunk 1]
```diff
@@ -76,6 +77,13 @@ private static void validateName(String name) {
         if (!VALID_EXTENSION_PATTERN.matcher(name).matches()) {
             throw new InvalidUserDataException("Invalid model name '" + name + "': it must match the following regular expression: " + VALID_EXTENSION_NAME);
         }
+
+        if (!name.equals("libs") && !name.endsWith("Libs")) {
+            DeprecationLogger.deprecateBehaviour("The name of version catalogs must end with 'Libs' to reduce chances of extension conflicts.")
+                .willBeRemovedInGradle8()
```
Comment (octylFractal) [inline subprojects/dependency-management/src/main/java/org/gradle/internal/management/DefaultVersionCatalogBuilderContainer.java:83]: Isn't this `willBecomeAnErrorInGradle8()`?

**Review by octylFractal — CHANGES_REQUESTED** [review]
One comment about the deprecation warning.

Also, I think it would be good to test for the deprecation warning.

File: subprojects/dependency-management/src/main/java/org/gradle/internal/management/DefaultVersionCatalogBuilderContainer.java (hunk 1 above)
Comment (ljacomet) [reply subprojects/dependency-management/src/main/java/org/gradle/internal/management/DefaultVersionCatalogBuilderContainer.java]: 🤦

**Comment by ljacomet** [comment]
@bot-gradle test this

**Review by octylFractal — APPROVED** [review]

**Comment by ljacomet** [comment]
@bot-gradle test and merge

**Pull request merged**